openverse_catalog/dags/data_refresh/refresh_popularity_metrics_task_factory.py

```diff
@@ -0,0 +1,48 @@
+"""
+# Refresh Popularity Metrics TaskGroup Factory
+This file generates a TaskGroup that refreshes the underlying popularity DB
+tables, using a factory function.
+
+This step updates any changes to popularity metrics, and recalculates the
+popularity constants. It should be run at least once every month, or whenever
+a new popularity metric is added. Scheduling is handled in the parent data
+refresh DAG.
+"""
+import os
+
+from airflow.utils.task_group import TaskGroup
+from common.popularity import operators
+
+
+DB_CONN_ID = os.getenv("OPENLEDGER_CONN_ID", "postgres_openledger_testing")
+GROUP_ID = "refresh_popularity_metrics_and_constants"
+
+
+def create_refresh_popularity_metrics_task_group(media_type: str):
+    """
+    This factory method instantiates a TaskGroup that will update the popularity
+    DB tables for the given media type, including percentiles and popularity
+    metrics.
+
+    Required Arguments:
+
+    media_type:  the type of record to refresh
+    """
+    with TaskGroup(group_id=GROUP_ID) as refresh_all_popularity_data:
+        # Update the popularity metrics table, adding any new popularity metrics
+        # and updating the configured percentile.
+        update_metrics = operators.update_media_popularity_metrics(
+            DB_CONN_ID,
+            media_type=media_type,
+        )
```

Comment:
We can't quite do this here with the functional approach, but the [`BaseOperator` supports several variants of `doc_<x>` fields](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/baseoperator/index.html#airflow.models.baseoperator.BaseOperator). I think these comments would be perfect for that at the task level!

Reply:
I didn't realize this was available for Tasks! Very handy, especially because it was tricky writing a concise doc for the DAG overall.